### PR TITLE
Add Symbol.allMembers to reflection API

### PR DESCRIPTION
**Pull request by pikinier20**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
Part of #13371

File: library/src/scala/quoted/Quotes.scala [hunk 1]
```diff
@@ -3652,6 +3652,9 @@ trait Quotes { self: runtime.QuoteUnpickler & runtime.QuoteMatching =>
         /** All members directly declared in the class */
         def declarations: List[Symbol]
 
+        /** All members declared or inherited */
+        def allMembers: List[Symbol]
```
[the remark this answers is not in the document]
Comment (nicolasstucki) [reply library/src/scala/quoted/Quotes.scala:3760]: This method should also be marked as `@experimental`

File: compiler/src/scala/quoted/runtime/impl/QuotesImpl.scala [hunk 2]
```diff
@@ -2603,6 +2603,13 @@ class QuotesImpl private (using val ctx: Context) extends Quotes, QuoteUnpickler
         def declarations: List[Symbol] =
           self.typeRef.info.decls.toList
 
+        def allMembers: List[Symbol] =
+          lookupPrefix.allMembers.iterator.map(_.symbol).collect {
```
Comment (nicolasstucki) [inline compiler/src/scala/quoted/runtime/impl/QuotesImpl.scala:2607]: Based on the test I added, the order of the members is not deterministic.

**Comment by pikinier20** [comment]
For now, we'll stick to internal compiler API in terms of this method.

**Pull request closed without merging**